src/tel_download.js

```diff
@@ -12,10 +12,15 @@
 // @match        https://webk.telegram.org/*
 // @match        https://webz.telegram.org/*
 // @icon         https://img.icons8.com/color/452/telegram-app--v5.png
-// @grant        none
+// @grant        window.onurlchange
 // ==/UserScript==
 
 (function () {
+  let audio_links = new Map()
```

Comment:
Instead of using a global Map, could we simply checking the existance of the download button below?